File: frontend/src/types/session.ts
export const SessionStatus = {
  None: "none",
  Running: "running",
  Paused: "paused",
  Closed: "closed",
} as const;
export type SessionStatus = (typeof SessionStatus)[keyof typeof SessionStatus];

export interface Session {
  id: string;
  taskId: string;
  agent: "claude" | "codex";
  sessionId: string | null;
  status: SessionStatus;
  createdAt: string;
  lastActive: string;
}

export interface StartSessionResponse {
  sessionPk: string;
  taskId: string;
  sessionId: string | null;
  sessionIdMissing: boolean;
  status: SessionStatus;
  createdAt: string;
}

// Legacy export kept for backward compatibility
export const SessionState = {
  NoSession: "no-session",
  Active: "active",
  Resumable: "resumable",
  Closed: "closed",
} as const;
export type SessionState = (typeof SessionState)[keyof typeof SessionState];

File: backend/src/models/session.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct Session {
    pub id: String,
    pub task_id: String,
    pub agent: String,
    pub session_id: Option<String>,
    pub status: String,
    pub created_at: String,
    pub last_active: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct StartSessionResponse {
    pub session_pk: String,
    pub task_id: String,
    pub session_id: Option<String>,
    pub session_id_missing: bool,
    pub status: String,
    pub created_at: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct CancelSessionResponse {
    pub task_id: String,
    pub status: String,
    pub message: String,
}

#[derive(Debug, Deserialize, Default)]
#[serde(default, rename_all = "camelCase")]
pub struct StartSessionRequest {}
